lib/runtime/src/component/endpoint.rs

```diff
@@ -288,10 +286,19 @@ fn build_transport_type(
 
             // Include endpoint name for proper TCP routing
             // TCP client parses this format and adds x-endpoint-path header for server-side routing
-            let tcp_endpoint = format!("{}:{}/{}", tcp_host, tcp_port, endpoint_name);
+            let tcp_endpoint = format!("{}:{}/{}", tcp_host, tcp_port, endpoint_id.name);
 
             TransportType::Tcp(tcp_endpoint)
         }
-        RequestPlaneMode::Nats => TransportType::Nats(subject.to_string()),
+        RequestPlaneMode::Nats => TransportType::Nats(nats_subject(endpoint_id, instance_id)),
     }
 }
+
+/// The NATS subject to talk to this instance on.
+/// TODO: Do we need to sanitize the names?
+fn nats_subject(endpoint_id: &EndpointId, instance_id: u64) -> String {
```

Comment:
question: should this be in a nats specific transport file - and not in endpoint?

Reply:
Yes, I thought that too but didn't get to it. Now moved. Thanks for keeping me honest.